CRM/MembershipExtras/Form/MembershipPeriod/Base.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+use CRM_MembershipExtras_Service_ContributionUtilities as ContributionUtilities;
+
+/**
+ * Base for forms to update membership periods.
+ */
+abstract class CRM_MembershipExtras_Form_MembershipPeriod_Base extends CRM_Core_Form {
+
+  /**
+   * ID of the period to be activated.
+   *
+   * @var int
+   */
+  protected $id;
+
+  /**
+   * Maps status ID's to status names.
+   *
+   * @var array
+   */
+  protected $contributionStatusesValueMap;
+
+  /**
+   * @inheritdoc
+   */
+  public function preProcess() {
+    $this->id = CRM_Utils_Request::retrieve('id', 'String', $this, TRUE);
+    $this->contributionStatusesValueMap = ContributionUtilities::getStatusesValueMap();
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function buildQuickForm() {
+    $this->setFormTitle();
+
+    $this->addButtons([
+      [
+        'type' => 'submit',
+        'name' => E::ts('Activate'),
+        'isDefault' => TRUE,
+      ],
+      [
+        'type' => 'cancel',
+        'name' => ts('Cancel'),
+      ],
+    ]);
+
+    $period = $this->getMembershipPeriod();
```

Comment:
there is already a BAO method for this : https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/e12ced93712439210133a9e76b74a1ca4bea730a/CRM/MembershipExtras/BAO/MembershipPeriod.php#L56

It is better to use it instead of duplicating it.

Reply:
Used the BAO method.